### PR TITLE
Maryia/TS-Package-7/add types for SmartCharts components

### PR DESCRIPTION
**Pull request by maryia-deriv**
## Changes:

- to add types for SmartCharts components

Screenshot 2023-12-04 at 5 30 58 PM

### CODE REVIEW

File: packages/trader/src/Modules/SmartChart/index.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    xPositioner?: string;
+    yPositioner?: string;
+    overlap_y_axis?: boolean;
+}>;
+
+type TPropsWithPortalNodeId = {
+    portalNodeId?: string;
+};
+
+type TSmartChartProps = React.PropsWithChildren<{
```
Comment (maryia-deriv) [inline packages/trader/src/Modules/SmartChart/index.tsx:138]: types from the removed SmartChartSwitcher https://github.com/kate-deriv/deriv-app/pull/45/files#diff-04bb2774644da792dc593a913b0892dcd4d21c260415ddda81814bd041f85bb1

**Review by kate-deriv — APPROVED** [review]

File: packages/trader/src/Modules/SmartChart/index.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+type TLanguage = {
+    icon: JSX.Element;
+    key: string;
+    name: string;
+};
+
+type TNotification = {
+    category: string;
+    text: string;
+    type?: 'info' | 'warning' | 'success' | 'error' | 'warning';
```
Comment (henry-deriv) [inline packages/trader/src/Modules/SmartChart/index.tsx:43]: extra 'warning'

**Review by henry-deriv — APPROVED** [review]

**Review by balakrishna-deriv — APPROVED** [review]

File: packages/trader/src/Modules/SmartChart/index.tsx (hunk 2 above)
Comment (kate-deriv) [reply packages/trader/src/Modules/SmartChart/index.tsx:43]: I will merge this PR into the parent branch and will remove extra 'warning'

Comment (maryia-deriv) [reply packages/trader/src/Modules/SmartChart/index.tsx:43]: thank you @kate-deriv :)

**Pull request merged**